modular_skyrat/master_files/code/game/objects/structures/sauna_oven.dm

```diff
@@ -98,15 +99,30 @@
 /obj/structure/sauna_oven/process()
 	if(water_amount)
 		water_amount--
+		update_steam_particles()
 		var/turf/open/pos = get_turf(src)
 		if(istype(pos) && pos.air.return_pressure() < 2*ONE_ATMOSPHERE)
 			pos.atmos_spawn_air("water_vapor=10;TEMP=[SAUNA_H2O_TEMP]")
 	fuel_amount--
 	if(fuel_amount <= 0)
 		lit = FALSE
+		update_steam_particles()
 		STOP_PROCESSING(SSobj, src)
 		update_icon()
 
+/obj/structure/sauna_oven/proc/update_steam_particles()
+	if(particles)
+		if(lit && water_amount)
+			return
+		else
+			QDEL_NULL(particles)
+			return
+	if(!particles)
+		if(lit && water_amount)
+			particles = new /particles/smoke/steam/mild
+			particles.position = list(0, 6, 0)
+			return
+
```

Comment:
```suggestion
/obj/structure/sauna_oven/proc/update_steam_particles()
	if(particles)
		if(lit && water_amount)
			return
		QDEL_NULL(particles)
		return

	if(lit && water_amount)
		particles = new /particles/smoke/steam/mild
		particles.position = list(0, 6, 0)

```

Reply:
yeah that looks like a better way to do it I think